src/handle_snapshot_sync.cxx

```diff
@@ -126,9 +126,9 @@ ptr<req_msg> raft_server::create_sync_snapshot_req(peer& p,
         // LCOV_EXCL_START
         // Raw binary snapshot (original)
         ulong offset = p.get_snapshot_sync_ctx()->get_offset();
-        int32 sz_left = (int32)(snp->size() - offset);
+        ulong sz_left = snp->size() - offset;
```

Comment:
@greensky00 I think here we can guarantee that `snp->size() >= offset`?
Or should we use `ulong sz_left = ( ulong sz_left = snp->size() - offset ) ? ( snp->size() - offset ) : 0`?

Reply:
Yes, please go ahead, `sz_left` also should not overflow.